blocks/api/utils.js

```diff
@@ -66,7 +83,7 @@ export function normalizeIconObject( icon ) {
 	if ( ! icon ) {
 		return { src: 'block-default' };
 	}
-	if ( isString( icon ) || isFunction( icon ) || icon instanceof Component ) {
+	if ( isValidIcon( icon ) ) {
 		return { src: icon };
```

Comment:
Not required because if "icon" is a valid icon it is a string or a React/element/component, and in that case, it should not contain a background color.